locales/i18n.js

```diff
@@ -19,11 +19,20 @@ function formatLocale(lang) {
 function urlOverrideLang(path) {
   var localPath = path || location.pathname;
   var localeCode = localPath.split('/')[1];
-  var pathname = localPath.split('/')[2] || localeCode;
+  localeCode = (/^[a-z]{2,3}(-[A-Z]{2})?$/.test(localeCode)) ? localeCode : null;
```

Comment:
I'm familiar yes. But, this (i18n.js) is client side code, so we would just have to make a copy of that file, not just require it. 
Anyway, seems a bit heavy to me for a minor use like this when we don't really care about the details of the locale.